cpp/harvest/nacjd.cc

```diff
@@ -119,6 +135,17 @@ void HandleChar(const char c) {
     }
 }
 
+std::string CheckingStatisticsCategory(std::string data_title) {
+    std::string category_name("");
+    for (std::string n : statistics_categories_array) {
+        size_t pos = data_title.find(n);
+        if (pos != std::string::npos) {
+            category_name = n;
```

Comment:
You can simply return the category here directly to skip the rest of the iteration when the first category is found, and then just return an empty string as a fallback if no categories are found at the end of the function.